pkg/resource/api.go

```diff
@@ -124,19 +126,15 @@ func (a *APIManagedConnectionPropagator) PropagateConnection(ctx context.Context
 		}
 		cmcs.Data = mgcs.Data
 		meta.AddAnnotations(cmcs, map[string]string{
-			AnnotationKeyPropagateFromNamespace: mgcs.GetNamespace(),
-			AnnotationKeyPropagateFromName:      mgcs.GetName(),
-			AnnotationKeyPropagateFromUID:       string(mgcs.GetUID()),
+			fmt.Sprintf(AnnotationKeyPropagateFromFormat, string(mgcs.GetUID())): strings.Join([]string{mgcs.GetNamespace(), mgcs.GetName()}, "/"),
 		})
 		return nil
 	}); err != nil {
 		return errors.Wrap(err, errCreateOrUpdateSecret)
 	}
 
 	meta.AddAnnotations(mgcs, map[string]string{
-		AnnotationKeyPropagateToNamespace: cmcs.GetNamespace(),
-		AnnotationKeyPropagateToName:      cmcs.GetName(),
-		AnnotationKeyPropagateToUID:       string(cmcs.GetUID()),
+		fmt.Sprintf(AnnotationKeyPropagateToFormat, string(cmcs.GetUID())): strings.Join([]string{cmcs.GetNamespace(), cmcs.GetName()}, "/"),
```

Comment:
Perhaps the `/` separator should be a constant, given it's used across at least two files.